graylog2-web-interface/src/components/rules/RuleHelper.jsx

```diff
@@ -170,7 +179,11 @@ end`,
                 </Row>
                 <Row>
                   <Col sm={12}>
-                    <FilterInput onChange={this._filterDescriptors} />
+                    <SearchForm onSearch={this._filterDescriptors}
+                                label="Filter rules"
+                                topMargin={0}
+                                searchButtonLabel="Filter"
+                                onReset={this._onFilterReset} />
                     <div className={`table-responsive ${RuleHelperStyle.marginTab}`}>
                       <PaginatedList totalItems={functionDescriptors.length} pageSize={this.state.pageSize} onChange={this._onPageChange} showPageSizeSelect={false}>
```

Comment:
I don't think is very important, or if it's related to this change, but the save/reset page is not working as I expected. Examples:

1. Go to page 3 of functions
2. Filter `is`, as that's the function I want to use
3. The pagination says I'm in page 3, but I am actually in page 1

--

1. Filter items with `is`
2. Go to page 2
3. Reset the filter
4. The pagination says I'm in page 2, but I'm in page 1

---

After looking at the code a bit, I think all we are missing is to pass the `activePage` prop to `PaginatedList`. That should make the pager change the page when changed externally.